### PR TITLE
net: openthread: allow to configure platform info

### PR DESCRIPTION
**Pull request by edmont**
This commit adds the option to configure the platform information string of OpenThread.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

### CODE REVIEW

File: subsys/net/l2/openthread/Kconfig.thread [hunk 1]
```diff
@@ -55,10 +55,14 @@ config OPENTHREAD_POLL_PERIOD
 
 config OPENTHREAD_MAX_CHILDREN
 	int "The maximum number of children"
-	range 10 512
+	range 10 511
 	default 32
 
 config OPENTHREAD_MAX_IP_ADDR_PER_CHILD
-	int "The maximum number of IPv6 address registrations per child."
+	int "The maximum number of IPv6 address registrations per child"
```
Comment (jukkar) [inline subsys/net/l2/openthread/Kconfig.thread:62]: Looks like these changes should be in some other PR, as the commit talks only about configuring platform info.

Comment (edmont) [reply subsys/net/l2/openthread/Kconfig.thread:62]: I updated the commit description.

Comment (jukkar) [reply subsys/net/l2/openthread/Kconfig.thread:62]: Thanks. The changes look unrelated, could you split the commit to two, so this PR would have two commits for these changes.

Comment (edmont) [reply subsys/net/l2/openthread/Kconfig.thread:62]: Done.

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**